src/Symfony/Component/Security/Http/Firewall/ContextListener.php

```diff
@@ -92,7 +96,21 @@ public function __invoke(RequestEvent $event)
         $request = $event->getRequest();
         $session = $request->hasPreviousSession() && $request->hasSession() ? $request->getSession() : null;
 
-        if (null === $session || null === $token = $session->get($this->sessionKey)) {
+        if (null !== $session) {
+            $usageIndexValue = method_exists(Request::class, 'getPreferredFormat') && $session instanceof Session ? $usageIndexReference = &$session->getUsageIndex() : 0;
```

Comment:
this makes one component rely on internal APIs of another component, which is not covered by semver. This looks like a bad idea, as our composer constraints are relying on semver (we got some similar issues in the past when using semver and relying on APIs not covered by our BC promise).
A change in HttpFoundation would then be able to break previous releases of Security.

Reply:
That's why it's tested. I don't think we should make anything non-internal here.

Reply:
Note that my opinion here is based on the idea that the `Session` namespace is underdesigned right now. Extending it would make no sense when it needs a fresh redesign from the ground up. Until this happens, let's at least make things work correctly.

Reply:
You won't be testing *existing releases of symfony/security-http* against *future changes of security/http-foundation*. CI will ensure that new versions of symfony/security-http will work fine, not that existing releases of symfony/security-http will keep working fine. What ensures that they keep working fine is the BC promise... except it does not cover this.
If we need some cross-component usage of the session usage tracking system, we need to upgrade it to a non-internal API, or we need to change symfony/security-http to use an exact-match requirement to `self.version` (which is quite bad news).

Reply:
Both suggestions are no-go as we together give arguments as to why each are. `getPreferredFormat` is covered by the BC promise, so the behavior is going to be activated and won't disappear unnoticed. I'm not for building complexity to guard against an hypothetical future, especially when there is this hint that can't be missed in that future.

Reply:
As discussed on Slack, I removed the `@internal` annotation on `Request::getUsageIndex()` so that we're forced to handle it with the BC promise.